openpaas-james/mailbox/encrypted/cassandra/src/main/java/com/linagora/openpaas/encrypted/cassandra/CassandraKeystoreDAO.scala

```diff
@@ -0,0 +1,74 @@
+package com.linagora.openpaas.encrypted.cassandra
+
+import java.nio.ByteBuffer
+
+import com.datastax.driver.core.querybuilder.QueryBuilder
+import com.datastax.driver.core.querybuilder.QueryBuilder.{asc, bindMarker, delete, insertInto, select}
+import com.datastax.driver.core.{PreparedStatement, Row, Session}
+import com.linagora.openpaas.encrypted.cassandra.table.KeyStoreTable.{ID, KEY, TABLE_NAME, USERNAME}
+import com.linagora.openpaas.encrypted.{KeyId, PublicKey}
+import javax.inject.Inject
+import org.apache.james.backends.cassandra.init.CassandraTypesProvider
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor
+import org.apache.james.core.Username
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class CassandraKeystoreDAO @Inject()(executor: CassandraAsyncExecutor,
+                                     insertStatement: PreparedStatement,
+                                     selectStatement: PreparedStatement,
+                                     selectAllStatement: PreparedStatement,
+                                     deleteStatement: PreparedStatement,
+                                     deleteAllStatement: PreparedStatement) {
+
+  def this(session: Session, cassandraTypesProvider: CassandraTypesProvider) {
+    this(
+      executor = new CassandraAsyncExecutor(session),
+      insertStatement = session.prepare(insertInto(TABLE_NAME)
+        .value(USERNAME, bindMarker(USERNAME))
+        .value(ID, bindMarker(ID))
+        .value(KEY, bindMarker(KEY))),
+
+      selectStatement = session.prepare(select.from(TABLE_NAME)
+        .where(QueryBuilder.eq(USERNAME, bindMarker(USERNAME)))
+        .and(QueryBuilder.eq(ID, bindMarker(ID)))
+        .allowFiltering),
```

Comment:
`allowFiltering` 

Euhhhh are you sure????

Reply:
Cassandra will throw without allowFiltering option. USERNAME is PRIMARY KEY and the query is filtering on ID field, no? May be we need to define a COMPOSITE KEY (USERNAME + ID)?

Reply:
ID being a clustering column you should not have this behaviour. If you have it then you are doing something wrong.